src/RegistrationController.js

```diff
@@ -112,16 +112,49 @@ export default BaseLoginController.extend({
           }
         );
       })
-      .fail(function(err) {
+      .fail((err) => {
         const responseJSON = err.responseJSON;
 
         if (responseJSON && responseJSON.errorCauses.length) {
-          const errMsg = responseJSON.errorCauses[0].errorSummary;
+          const { reason } = responseJSON.errorCauses[0];
 
-          Util.triggerAfterError(self, new Errors.RegistrationError(errMsg));
+          const isNotUniqueValue =
+            responseJSON.errorCode === 'E0000001' &&
+            reason === 'UNIQUE_CONSTRAINT';
+
+          if (isNotUniqueValue) {
+            this.renderIsNotUniqueError(responseJSON);
+          }
+
+          const errMsg = responseJSON.errorSummary;
+          Util.triggerAfterError(this, new Errors.RegistrationError(errMsg));
         }
       });
   },
+
+  renderIsNotUniqueError: function(error) {
+    // remove generic form-level error
+    this.model.trigger('form:clear-errors');
+
+    error.errorSummary = loc(
+      'registration.error.account.notUniqueWithinOrg',
+      'login',
+      [error.errorCauses[0].location]
+    );
+    setTimeout(
+      () =>
+        this.model.trigger(
+          'error',
+          this.model,
+          {
+            responseJSON: { errorSummary: error.errorSummary }, // remove errorCauses so message renders at form-level
+          },
+          true
+        ),
+      100 // to beat race-condition with something that is clearing this error message
```

Comment:
I couldn't track down what was clearing. I listened to 'form:clear-errors' on the model and nothing seems to be calling it.

Reply:
I'd like to find a way to remove this `setTimeout`

Reply:
There's a race condition somewhere. If I have breakpoints set, I don't need this setTimeout.

Reply:
Race condition demo: https://okta.box.com/s/qbn4rb82ojiyb84a0b1mvi9rqh7nx33p